src/vaadin-message.js

```diff
@@ -151,17 +152,46 @@ class MessageElement extends ElementMixin(ThemableMixin(PolymerElement)) {
     `;
   }
 
-  ready() {
-    super.ready();
-  }
-
   static get is() {
     return 'vaadin-message';
   }
 
   static get version() {
     return '2.0.0-alpha1';
   }
+
+  ready() {
+    super.ready();
+
+    // Handle focus
+    this.addEventListener('focus', () => this._setFocused(true), true);
+    this.addEventListener('blur', () => this._setFocused(false), true);
+    this.addEventListener('mousedown', () => {
+      this._mousedown = true;
+      const mouseUpListener = () => {
+        this._mousedown = false;
+        document.removeEventListener('mouseup', mouseUpListener);
+      };
+      document.addEventListener('mouseup', mouseUpListener);
+    });
+  }
+
+  /**
+   * @param {boolean} focused
+   * @protected
+   */
+  _setFocused(focused) {
+    if (focused) {
+      this.setAttribute('focused', '');
+      if (!this._mousedown) {
+        this.setAttribute('focus-ring', '');
+      }
+      this.dispatchEvent(new CustomEvent('messagefocus', { detail: { message: this } }));
```

Comment:
Why is this event needed? It should be possible to use `focusin` event in `vaadin-message-list` instead.

Reply:
Do you mean something like this in `vaadin-message-list.js ready()`:
```js
this.addEventListener('focusin', (event) => {
  const message = event.composedPath()[0];
  this._setTabIndexesByMessage(message);
});
```
It works for the first click, and you get the message that was clicked. But if you click a second time on another message, it doesn't fire an event as focus is still in messageList. Therefore I can't update the tabindexes so that the last message clicked would have 0. Any ideas on how to proceed?

Reply:
Oh I think I can change the template to be 
```html
<vaadin-message on-focusin="_handleFocusEvent"
```

Reply:
```html
on-focusin="_handleFocusEvent"
``` 
worked when manually testing, but the listener did not get triggered by the tests when doing

```
message.dispatchEvent(new CustomEvent('focus', { composed: true, bubbles: true }));
```

However, when listening to `focus` instead of `focusin` it worked.
```html
on-focus="_handleFocusEvent"
```
I put a commit in with this change. Is there some problems with using `focus` instead of `focusin`?